src/main/java/org/zalando/nakadi/service/subscription/SubscriptionService.java

```diff
@@ -227,25 +233,39 @@ public Result listSubscriptions(final String owningApplication, final Set<String
     private Stream<SubscriptionEventTypeStats.Partition> filterTopicPartitions(
             final ZkSubscriptionClient zkSubscriptionClient,
             final List<TopicPartition> topicPartitions,
-            final Partition partition) {
+            final Partition partition,
+            final boolean hasSessions) {
         return topicPartitions.stream()
                 .filter(topicPartition ->
                         partition.getKey().getPartition().equals(topicPartition.getPartitionId()))
                 .map(Try.wrap(topicPartition ->
-                        createPartition(zkSubscriptionClient, partition, topicPartition)))
+                        createPartition(zkSubscriptionClient, partition, topicPartition, hasSessions)))
                 .map(Try::getOrThrow);
     }
 
     private SubscriptionEventTypeStats.Partition createPartition(final ZkSubscriptionClient zkSubscriptionClient,
                                                                  final Partition partition,
-                                                                 final TopicPartition topicPartition)
+                                                                 final TopicPartition topicPartition,
+                                                                 final boolean hasSessions)
             throws NakadiException {
         final String partitionName = partition.getKey().getPartition();
-        final String partitionState = partition.getState().getDescription();
-        final String partitionSession = partition.getSession();
+        final String partitionState;
+        final String partitionSession;
+        if (hasSessions){
+            partitionState = partition.getState().getDescription();
+            partitionSession = partition.getSession();
+        } else {
+            partitionState = Partition.State.UNASSIGNED.getDescription();
+            partitionSession = "";
+        }
+
         final long clientOffset = zkSubscriptionClient.getOffset(partition.getKey());
-        final long total = Long.valueOf(topicPartition.getNewestAvailableOffset());
-        final long unconsumedEvents = total - clientOffset;
+        Long unconsumedEvents = null;
```

Comment:
@v-stepanov I will fix it.
@rcillo According to API https://github.com/zalando/nakadi/blob/nakadi-jvm/api/nakadi-event-bus-api.yaml#L1523 we have to hide this field when it was not read yet.